ImageLoadingProject/ImageLoadingProject/LargeImageViewController.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  LargeViewViewController.swift
+//  ImageLoadingProject
+//
+//  Created by Nikita Sosyuk on 26.11.2020.
+//
+
+import UIKit
+
+class LargeImageViewController: UIViewController, URLSessionDownloadDelegate {
+    
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var progressView: UIProgressView!
+    
+    var urlString: URL?
+    private var imageView: UIImageView?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loadImage()
+        progressView.progress = 0
+    }
+    
+    public func loadImage() {
+        guard let url = urlString else { return }
+        let urlSession = URLSession(configuration: .default, delegate: self, delegateQueue: OperationQueue())
+        let downloadTask = urlSession.downloadTask(with: url)
+        downloadTask.resume()
+    }
+    
+    func urlSession(_ session: URLSession, downloadTask: URLSessionDownloadTask, didFinishDownloadingTo location: URL) {
+        if let data = try? Data(contentsOf: location) {
+            DispatchQueue.main.async { [weak self] in
+                self?.scrollView.addSubview(UIImageView(image: UIImage(data: data)))
+                self?.progressView.isHidden = true
```

Comment:
Можно было доделать, чтобы заработало
```suggestion
                guard let scrollView = self?.scrollView, let image = UIImage(data: data) else { return }
                self?.progressView.isHidden = true
                let imageView = UIImageView(image: image)
                imageView.frame.size = image.size
                scrollView.addSubview(imageView)
                NSLayoutConstraint.activate([
                    scrollView.topAnchor.constraint(equalTo: imageView.topAnchor),
                    scrollView.bottomAnchor.constraint(equalTo: imageView.bottomAnchor),
                    scrollView.leadingAnchor.constraint(equalTo: imageView.leadingAnchor),
                    scrollView.trailingAnchor.constraint(equalTo: imageView.trailingAnchor),
                ])
```